safekeeper/src/timeline.rs

```diff
@@ -663,7 +664,9 @@ impl Timeline {
 
             // if this is AppendResponse, fill in proper hot standby feedback.
             if let Some(AcceptorProposerMessage::AppendResponse(ref mut resp)) = rmsg {
-                resp.hs_feedback = self.walsenders.get_hotstandby();
+                let standby_feedback = self.walsenders.get_hotstandby();
+                resp.hs_feedback = standby_feedback.hs_feedback;
+                shared_state.sk.state.inmem.standby_horizon = standby_feedback.reply.apply_lsn;
```

Comment:
This is a quite random place to copy standby_horizon from Walsenders to Timeline state. In particular, it won't work at all if primary is down. Let's remove inmem.standby_horizon completely and instead get standby_horizon from Walsenders directly when needed?

Reply:
I agree that it seems to be quite random place.
Information about standby apply lsn is needed in the single place: `SharedState::get_safekeeper_info()`.
There is no reference to timeline (to call `walsenders.get_hotstandby()`) but there is definitely reference to `self.sk.state.inmem` and it is used here to get `peer_horizon`, `remote_consistent_lsn` and many other stuff.
So I do not think why it "it won't work at all if primary is down.".

In any case - you are much ore family,iar with this code than I. Please move or remove standby_horizon  where you find art most appreciate.

Reply:
done